YoutubeExplode/Internal/YoutubeVideoStream.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace YoutubeExplode.Internal
+{
+    internal class YoutubeVideoStream : Stream
+    {
+        private readonly HttpClient _httpClient;
+
+        private const int MaxSegmentSize = 9_898_989;
+
+        private readonly string _url;
+
+        Stream _currentStream;
+
+        private long _position;
+
+        public YoutubeVideoStream(HttpClient httpClient, string url, long length)
+        {
+            _url = url;
+            _httpClient = httpClient;
+            Length = length;
```

Comment:
Is `Length` required? Can we avoid having to pass it so that it's more versatile?

Reply:
Unfortunately we need that Length, otherwise I don't know how much youtube can give me, if I go over the boundaries `await _httpClient.GetStreamAsync(_url, Length, Length + MaxSegmentSize)` (basically when Position > Length) youtube returns 400 invalid range requested (unless you want me to catch that exception instead and say that stream is done).

I could try calculating if youtube doesn't return stream of size `MaxSegmentSize` as I request, I could say that this is the last stream they will give, but if MaxSegmentSize is exactly the size of a video, or if youtube decides to give me smaller stream then I requested anyway, it would just make it more brittle in my opinion.

Reply:
And unfortunately response headers do not return overall size, only `Content-Length` which is for current request which in our case it's `MaxSegmentSize` most of the time for us.

Reply:
Fair enough